ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -361,8 +361,13 @@ forkBlockProduction IS{..} =
           ChainDB.addBlock chainDB newBlock
           -- Check whether we adopted our block
           curTip <- atomically $ ChainDB.getTipPoint chainDB
+          -- We measure the time directly.
+          -- We could also measure it when collecting the trace events, but this
+          -- is much more precise.
+          time <- getMonotonicTime
```

Comment:
I would move this to the `then` branch, because we only use it there. It doesn't really matter much, though.